src/nupic/test/abstract_temporal_memory_test.py

```diff
@@ -20,26 +20,41 @@
 # ----------------------------------------------------------------------
 
 import unittest
+from abc import ABCMeta, abstractmethod
 
 from nupic.data.generators.sequence_machine import SequenceMachine
 from nupic.research.monitor_mixin.temporal_memory_monitor_mixin import (
   TemporalMemoryMonitorMixin)
-from nupic.research.temporal_memory import TemporalMemory
 
 
 
-class AbstractTemporalMemoryTest(unittest.TestCase):
+class AbstractTemporalMemoryTest(object):
```

Comment:
@mrcslws @scottpurdy 

This change is problematic for those of us who editors/IDEs with tab-completion. After this change, tab-completion on `unittest.TestCase` methods in classes like `TutorialTemporalMemoryTest` will no longer work. Perhaps renaming the files to replace the word "test" in the filename with something else or whatever it is that py.test is looking for (can be configured) would be a better option.